src/script/components/RichTextEditor/plugins/TypeaheadMenuPlugin.tsx

```diff
@@ -676,13 +663,6 @@ export function TypeaheadMenuPlugin<TOption extends TypeaheadOption>({
   const [editor] = useLexicalComposerContext();
   const [resolution, setResolution] = useState<Resolution | null>(null);
   const [menuVisible, setMenuVisible] = useState(false);
-  const anchorElementRef = useMenuAnchorRef({
-    containerId,
-    resolution,
-    setResolution,
-    className: `typeahead-menu ${anchorClassName || ''}`,
-    menuVisible,
-  });
```

Comment:
moving the anchor to a child component makes sure it's unmounted when there are no options to display